### ISSUE TITLE
Improve help message of `from_over_into` lint

### ISSUE THREAD

**Issue by flip1995** [rust-lang/rust-clippy#7088]
```rust
impl Into<c_int> or Signal {
...
}
```

In this case [...] the fix is to change last impl to this:

```rust
impl From<Signal> for c_int {
...
}
```

It might be helpful if the recommendation actually printed the type so it was explicitly obvious (implied). Here is an example error:

```
warning: an implementation of `From` is preferred since it gives you `Into<_>` for free where the reverse isn't true
   --> /mnt/host/source/src/platform/crosvm/sys_util/src/signal.rs:174:1
    |
174 | impl Into<c_int> for Signal {
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
    = help: consider to implement `From` instead
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#from_over_into
```

A better [help] message might look something like:

```
    = help: consider to implement `From<Signal>` instead
```

_Originally posted by @Allen-Webb in https://github.com/rust-lang/rust-clippy/issues/4894#issuecomment-820446092_

So the help message should at least mention the generic type that should be used. Maybe even produce an non-autoapplicable suggestion.

**Comment by Allen-Webb** [comment]
Just today I converted a few cases caught by the linter. One was even for a proc_macro. I think an autoapplicable suggestion might be possible but one hurdle would be coming up with an identifier `$ident` for the argument of `from()` that doesn't collide with any of the contents of the previous `into(self)` implementation. You would then replace self with `$ident`.

It seems like there may be some rare odd cases where `From<>` probably wouldn't be that useful such as for tuples that aren't given a named type and the developer might just silence the warning with an `#[allow()]`, but even in those cases automatically converting them wouldn't be harmful.

Here are links the the gerrit CLs where I made the changes in case they are helpful cases when developing the help message:
* [typical cases](https://chromium-review.googlesource.com/c/chromiumos/platform/crosvm/+/2827974/1/sys_util/src/descriptor.rs)
* [proc-macro case](https://chromium-review.googlesource.com/c/chromiumos/platform/crosvm/+/2826414/1/data_model/src/endian.rs)
* [not as useful case](https://chromium-review.googlesource.com/c/chromiumos/platform2/+/2826138/1/sirenia/libsirenia/src/transport/mod.rs#238)
